### DISCUSSION TITLE
How to create a discriminated union type with duplicate union choice

### DISCUSSION THREAD

**Discussion by yanyongyu** [pydantic/pydantic#8217] [Question]
This is an small example:

```python
from typing import Union, Literal, Annotated

from pydantic import Field, BaseModel, TypeAdapter


class A(BaseModel):
    type: Literal["a"] = "a"


class BOneOf0(BaseModel):
    type: Literal["b"] = "b"
    x: int


class BOneOf1(BaseModel):
    type: Literal["b"] = "b"
    y: int


MyType = Annotated[Union[A, Union[BOneOf0, BOneOf1]], Field(discriminator="type")]

TypeAdapter(MyType).validate_python({"type": "a"})
```

Pydantic v2 raises the following error (success in pydantic v1):

```python
TypeError: Value 'b' for discriminator 'type' mapped to multiple choices
```

I have tried using annotated `Tag`, but got the same error:

```python
MyType = Annotated[
    Union[A, Annotated[Union[BOneOf0, BOneOf1], Tag("b")]], Field(discriminator="type")
]
```

How can i parse the data into the type in this case?

**Comment by indepndnt** [comment]
I think you would want a callable discriminator that checks whether the model has the `x` or the `y` attribute. https://docs.pydantic.dev/latest/concepts/unions/#discriminated-unions-with-callable-discriminator
Unless you can change the `type` of `BOneOf0` or `BOneOf1`. It can't discriminate by that field because they are the same.

**Reply by yanyongyu** [reply]
Currently, i modified the schema of the union type to support pydantic union parsing mode:

https://github.com/yanyongyu/githubkit/blob/67a54c7e35df231beb3ec615acd466a33cacd05c/githubkit/utils.py#L71-L99
